lib/core/utils.js

```diff
@@ -278,42 +278,47 @@ RESUtils.isReddit = function() {
 	return RESUtils.regexes.all.test(currURL) && !RESUtils.regexes.toolbar.test(currURL);
 };
 RESUtils.pageType = function() {
-	if (typeof this.pageTypeSaved === 'undefined') {
-		var pageType = '';
-		var currURL = location.href.split('#')[0];
-		if (RESUtils.regexes.wiki.test(currURL)) {
-			pageType = 'wiki';
-		} else if (RESUtils.regexes.search.test(currURL)) {
-			pageType = 'search';
-		} else if (RESUtils.regexes.profile.test(currURL)) {
-			pageType = 'profile';
-		} else if ((RESUtils.regexes.comments.test(currURL)) || (RESUtils.regexes.friendsComments.test(currURL)) || (RESUtils.regexes.nosubComments.test(currURL))) {
-			pageType = 'comments';
-		} else if (RESUtils.regexes.inbox.test(currURL)) {
-			pageType = 'inbox';
-		} else if (RESUtils.regexes.submit.test(currURL)) {
-			pageType = 'submit';
-		} else if (RESUtils.regexes.prefs.test(currURL)) {
-			pageType = 'prefs';
-		} else if (RESUtils.regexes.stylesheet.test(currURL)) {
-			pageType = 'stylesheet';
-		} else {
-			pageType = 'linklist';
-		}
-		this.pageTypeSaved = pageType;
+	if (this.pageTypeSaved !== undefined) {
+		return this.pageTypeSaved;
+	}
+
+	var currURL = location.href.split('#')[0];
+	var pageType;
+	if (RESUtils.regexes.wiki.test(currURL)) {
+		pageType = 'wiki';
+	} else if (RESUtils.regexes.search.test(currURL)) {
+		pageType = 'search';
+	} else if (RESUtils.regexes.profile.test(currURL)) {
+		pageType = 'profile';
+	} else if (RESUtils.regexes.comments.test(currURL)) {
+		pageType = 'comments';
+	} else if (RESUtils.regexes.inbox.test(currURL)) {
+		pageType = 'inbox';
+	} else if (RESUtils.regexes.submit.test(currURL)) {
+		pageType = 'submit';
+	} else if (RESUtils.regexes.prefs.test(currURL)) {
+		pageType = 'prefs';
+	} else if (RESUtils.regexes.stylesheet.test(currURL)) {
+		pageType = 'stylesheet';
+	} else {
+		pageType = 'linklist';
 	}
-	return this.pageTypeSaved;
+
+	this.pageTypeSaved = pageType;
+	return pageType;
 };
 RESUtils.isPageType = function(/*type1, type2, type3, ...*/) {
-	var thisPage = RESUtils.pageType();
-	return Array.prototype.slice.call(arguments).some(function(e) {
-		return (e === 'all') || (e === thisPage);
+	var href = document.location.href;
+	return Array.prototype.slice.call(arguments).some(function(type) {
+		return (type in RESUtils.regexes &&
+			RESUtils.regexes[type].test(href)) ||
+			type === 'linklist';
```

Comment:
Eh, I can't see this being anything but slower: `isPageType()` gets called at least once per module, so it'll end up doing way more regex tests than the previous implementation.

Otherwise this is just a slightly more complex way to do it, so I don't think `isPageType()` needs to be changed. (If nothing else, this implementation matches `linklist` on every page, so that needs to be fixed.)